server/channels/app/post.go

```diff
@@ -1362,7 +1362,7 @@ func (a *App) GetPostsForChannelAroundLastUnread(c request.CTX, channelID, userI
 }
 
 func (a *App) DeletePost(c request.CTX, postID, deleteByID string) (*model.Post, *model.AppError) {
-	post, err := a.Srv().Store().Post().GetSingle(postID, false)
+	post, err := a.Srv().Store().Post().GetSingle(sqlstore.RequestContextWithMaster(c), postID, false)
```

Comment:
This is the actual change. Everything else is just to pass around the context.

Reply:
Out of curiosity, what was the purpose of `c request.CTX` serving without all these supporting changes? Is `c request.CTX` sometimes `nil`?

Reply:
It must have been passed to some other store method. Currently, there's an HW campaign to add context to all store methods even if they won't be used.